### PR TITLE
Deduplicate Kyber files

### PR DESCRIPTION
**Pull request by dkostic**
### Issues:
N/A

### Description of changes: 
The commit removes the file duplication issue with Kyber implementation.
Previously, Kyber512 and Kyber768 implementations resided in separate
folders, both with a copy of all the source files. In this change we:
 - create a single _common_ folder with Kyber source code files,
 - add `kyber512_ref.c` file that includes the required C files
   from the common folder and selects the Kyber512 implementation
   by defining a macro (`KYBER_K 2`) specified in Kyber as the selector
   of the variant to be compiled.

In this way, when we add Kyber768 support we will only have to add the
corresponding `kyber768_ref.c` file, which will avoid duplicating the
Kyber source code files.

### Call-outs:
Point out areas that need special attention or support during the review process. Discuss architecture or design changes.

### Testing:
How is this change tested (unit tests, fuzz tests, etc.)? Are there any testing steps to be verified by the reviewer?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and 
the ISC license.

### CODE REVIEW

File: crypto/kyber/kyber512_ref.c [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+#include "./pqcrystals_kyber_ref_common/cbd.h"
+#include "./pqcrystals_kyber_ref_common/fips202.h"
+#include "./pqcrystals_kyber_ref_common/indcpa.h"
+#include "./pqcrystals_kyber_ref_common/kem.h"
+#include "./pqcrystals_kyber_ref_common/ntt.h"
+#include "./pqcrystals_kyber_ref_common/poly.h"
+#include "./pqcrystals_kyber_ref_common/polyvec.h"
+#include "./pqcrystals_kyber_ref_common/reduce.h"
+#include "./pqcrystals_kyber_ref_common/symmetric.h"
+#include "./pqcrystals_kyber_ref_common/verify.h"
```
Comment (andrewhop) [inline crypto/kyber/kyber512_ref.c:20]: Do we need to include all the header files? Won't each c file include what it needs?

File: crypto/kyber/kem_kyber.h [hunk 2]
```diff
@@ -7,7 +7,7 @@
 #include <openssl/evp.h>
```
Comment (andrewhop) [inline crypto/kyber/kem_kyber.h:7]: Missing copyright, is this our file or kyber's originally?

**Review by andrewhop — COMMENTED** [review]
Also missing a copyright in p_kyber.c, p_kyber512_asn1.c

File: crypto/kyber/kyber512_ref.c (hunk 1 above)
Comment (dkostic) [reply crypto/kyber/kyber512_ref.c]: you are right, I removed the unnecessary includes.

File: crypto/kyber/kem_kyber.h (hunk 2 above)
Comment (dkostic) [reply crypto/kyber/kem_kyber.h:14]: added. it's our file

**Review by andrewhop — DISMISSED** [review]

File: crypto/kyber/kyber512_ref.c [hunk 3]
```diff
@@ -0,0 +1,23 @@
+/*
```
Comment (torben-hansen) [inline crypto/kyber/kyber512_ref.c:1]: okay, this does avoid dynamic and/or over-allocating stack memory etc in each function. But still "funny" macrofied.

**Review by torben-hansen — DISMISSED** [review]

**Review by torben-hansen — APPROVED** [review]

**Review by andrewhop — APPROVED** [review]

**Pull request merged**